### PR TITLE
feat(board): #MAG-21 add reading mode

### PR DESCRIPTION
**Pull request by Sofianebernoussi**
## Describe your changes

- add reading mode

## Checklist tests

- check if the cards are correctly displayed
- check if we can see the cards until the end
- check if the order of the cards is correct

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-21

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [X] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java [hunk 1]
```diff
@@ -174,7 +174,12 @@ public Future<JsonObject> getCard(String cardId) {
                 promise.fail(message);
                 return;
             }
-            promise.complete(results.right().getValue());
+            setMetadataCards(java.util.Collections.singletonList(new Card(results.right().getValue()))).onFailure(fail -> {
```
Comment (Sofianebernoussi) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java:177]: Cette façon de faire sera changer avec la prochaine PR

Comment (BastienBenard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java:177]: Tu peux utiliser un import pour enlever le java.util de java.util.Collections.singletonList + retour a la linge avant le .onFailure

Comment (Sofianebernoussi) [reply src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java:177]: prevu pour prochaine pr, je l'ai fait sur mon prochain ticket !

File: src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java [hunk 2]
```diff
@@ -174,7 +174,12 @@ public Future<JsonObject> getCard(String cardId) {
...
+                        log.error("[Magneto@%s::getAllCardsByBoard] Failed to get cards", this.getClass().getSimpleName(),
+                                fail.getMessage());
+                        promise.fail(fail.getMessage());
+                    })
+                    .onSuccess(card -> promise.complete(card.get(0).toJson()));
```
Comment (BastienBenard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java:182]: La variable crad ne peux jamais etre vide ?

File: src/main/resources/public/ts/controllers/board-read.controller.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        this.filter.page--;
+        await this.getCard();
+        safeApply(this.$scope);
+    }
+
+    getBoard = async (): Promise<void> => {
+        return this.boardsService.getBoardsByIds([this.filter.boardId])
+            .then((res: Boards) => {
+                if (!!res) {
+                    this.board = res.all[0];
```
Comment (BastienBenard) [inline src/main/resources/public/ts/controllers/board-read.controller.ts:89]: res.all ne peut pas être vide ?

**Review by flmartineau — APPROVED** [review]

**Pull request merged**